### PR TITLE
Define common types and CRD conversions

### PR DESCRIPTION
**Pull request by matt-mazzucato**
+ Implementing a conversion webhook would require to explicitly define the mapping between different CRD versions even if no changes occured. To avoid such a task, define common types which are shared between different CRD versions.
+ Implement a trivial conversion for CRDs v1alpha1/v1alpha2
+ Setup infrastructure for implementing admission/validation webhooks for Astarte, AVI, Flow.

### CODE REVIEW

File: apis/api/v1alpha1/conversion.go [hunk 1]
```diff
@@ -7,7 +7,7 @@
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
 
-     http://www.apache.org/licenses/LICENSE-2.0
+http://www.apache.org/licenses/LICENSE-2.0
```
Comment (rbino) [inline apis/api/v1alpha1/conversion.go:10]: Is this intentional?

**Review by rbino — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Expose port 8888 for VerneMQ metrics.
 - Added a Helm chart for deploying the Operator
 - Added Kubernetes Webhooks
+- Allow disabling webhooks with ENABLE_WEBHOOKS env var.
```
Comment (drf) [inline CHANGELOG.md:16]: Does it make sense allowing this in Helm, too? If so, the cert-manager dependency could be avoided in those cases

File: apis/api/v1alpha1/conversion.go [hunk 3]
```diff
@@ -19,4 +19,6 @@
 package v1alpha1
 
 // Hub marks this type as a conversion hub.
-func (*Astarte) Hub() {}
```
Comment (drf) [inline apis/api/v1alpha1/conversion.go:22]: I think all of this was done exactly to make v1alpha2 the Hub, isn't it?

**Review by drf — CHANGES_REQUESTED** [review]
I think the main missing point is inverting the Hub

File: CHANGELOG.md (hunk 2 above)
Comment (matt-mazzucato) [reply CHANGELOG.md:16]: In the long short run it should be ok. However if we want to rely on cert-manager on a wider scope, I wouldn't make the dependency optional.

**Review by rbino — APPROVED** [review]

**Review by drf — APPROVED** [review]

**Pull request merged**